pennylane/ops/op_math/sprod.py

```diff
@@ -191,8 +191,13 @@ def terms(self):  # is this method necessary for this class?
         Returns:
             tuple[list[tensor_like or float], list[.Operation]]: list of coefficients :math:`c_i`
             and list of operations :math:`O_i`
+
         """
-        return [self.scalar], [self.base]
+        try:
+            base_coeffs, base_ops = self.base.terms()
+            return [self.scalar * coeff for coeff in base_coeffs], base_ops
+        except TermsUndefinedError:
+            return [self.scalar], [self.base]
```

Comment:
out of curiosity, what is this case for? and is it already tested somewhere?

Reply:
This is essentially the base case of recursion, for example, the base of `3 * qml.X(0)` is `qml.X(0)`, and `qml.X(0).terms()` raises `TermsUndefinedError`.